### DISCUSSION TITLE
Creating Roadmaps

### DISCUSSION THREAD

**Discussion by jacobslusser** [jellyfin/jellyfin-meta#47] [General]
Hello Jellyfriends! I would like to help get the project organized, and of the ways is to create tasks that serve as roadmaps for other deliverables. For example, this is how Microsoft tracks feature delivery for .NET 8: [https://github.com/dotnet/aspnetcore/issues/44984](https://github.com/dotnet/aspnetcore/issues/44984).

An example for Jellyfin that I might suggest would be a roadmap for nonfunctional cleanup:

### Nonfunctional Cleanup
- [ ] Rename async methods with `Async` suffix
  - [ ] Use asnyc "all the way down"
- [ ] Move projects to `/src` folder
- [ ] Rename `Emby` to `Jellyfin`
- [ ] Introduce additional code analyzers
- [ ] Version API endpoints
- [ ] Consolidate helpers and utilities to a single project
- [ ] Use source generators for `ILogger` methods
- [ ] Mark classes `internal` which should not be public
- [ ] etc.

This is just an example and not everything on this list is necessarily applicable.

So, the question is, are the maintainers okay with me starting a few of these Roadmap issues, tagged with `roadmap` and start setting some goals? Is there a place to discuss any of this in private before taking that next step to make sure the goals are in sync?

Thanks for your consideration. I have personal goals for my code contributions, but looking to make sure we are all rowing in the same direction.

CC @Bond-009, @cvium, @Shadowghost, @thornbill

**Comment by cvium** [comment]
I like roadmaps and I like structure. I'm a developer and software architect by trade though and not a project manager. I hate making roadmaps and tasks. I don't mind following one.

I don't see much value in these nonfunctional cleanup though. It just breaks existing PRs, plugins etc. We should focus more on functional improvements and refactoring.

**Reply by oddstr13** [reply]
On the other hand, the sooner we do renaming cleanup, the fewer plugins will be affected – we're already late on that front tho, but that is a whole separate discussion.

Edit: and we should definitely try to do it all in a single release.

**Reply by jacobslusser** [reply]
Let me know what sort of refactoring you had in mind. I'm interested in taking on some of that work. It's a good way to get my bearings and improve the project at the same time.

**Reply by cvium** [reply]
It's practically everything. We need to refactor the scanner, the API, the database layer and the streambuilder. What we need more than anything is to break these tasks up somehow, but it's very difficult tbh

**Comment by Bond-009** [comment]
Roadmap issues aren't the right way to go about this IMO, I prefer individual issues and a project to collect them.
Our cleanup project for example: https://github.com/jellyfin/jellyfin/projects/14?query=is%3Aopen+sort%3Aupdated-desc

**Reply by jacobslusser** [reply]
I think this can work equally as well. It's just meant as a superset of individual tasks that are related so developers and see the project goals. If this can accomplish the same thing, great!

The one drawback is that I don't see a way in projects to have group discussions about the overall objectives -- just discussions within the individual tasks.

**Reply by oddstr13** [reply]
GitHub discussions like this one is good for discussions about larger architectural changes (groups of related issues/tasks), said discussion could be linked from the individual issues.
